Observer/Customer/Save.php

```diff
@@ -63,11 +63,17 @@ public function execute(Observer $observer)
         if ($customerAddress) {
             $data = array_merge($data, [
                 'country' => $customerAddress->getCountryId(),
-                'state' => $customerAddress->getRegion()->getRegionCode(),
+                'state' => '',
                 'zip' => $customerAddress->getPostcode(),
                 'city' => $customerAddress->getCity(),
                 'street' => implode(", ", $customerAddress->getStreet())
             ]);
+
+            if (get_class($customerAddress) == 'Magento\Customer\Model\Address') {
+                $data['state'] = $customerAddress->getRegionCode();
+            } elseif (get_class($customerAddress) == 'Magento\Customer\Model\Data\Address') {
+                $data['state'] = $customerAddress->getRegion()->getRegionCode();
+            }
```

Comment:
Are there any conditions here that would fall into an `else` we should be checking? Would this be an error state or require us to throw an exception to ensure we don't have a missing state field?

Reply:
@codymullins There are additional conditions I can think of (outside of a 3rd party extension that extends the Address model).  Ideally Magento's `AbstractAddress` and api data `AddressInterface` classes would be compatible (they both utilize the same data) but that isn't the case here.